src/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProvider.php

```diff
@@ -168,6 +176,13 @@ public function getDefaultData()
                 ],
                 'condition' => ProductCondition::NEW,
             ],
+            'categories' => [
+                'product_categories' => [
+                    $this->defaultCategoryId => [
+                        'is_associated' => true,
+                    ],
+                ],
```

Comment:
```suggestion
                ],
                'default_category' => $this->defaultCategoryId,
```

Reply:
I guess this suggestion fits only if I make form type for default category